js/jquery.flot.tooltip.js

```diff
@@ -80,117 +92,242 @@
+                    return;
+                } else if (self.tooltips[0].isSticky) {
+                    // If we have a sticky tooltip and one non-sticky and the user hovers over an
+                    // empty item, we want to hide the non-sticky tooltip only
+                    hideTooltip(1);
+                } else {
+                    hideTooltip(0);
+                }
+            } else if (tooltipCount === 1) {
+                if (self.tooltips[0].isSticky) {
+                    return;
+                } else {
+                    hideTooltip(0);
+                }
+            }
         }
 
+        /**
+         * Called whenever mouse hovers over a plot, ie, many times
+         * @param Object event jQuery event
+         * @param Object pos Contains positional (xy) data
+         * @param {Object} item Can be null if hovering over empty space. Config for a data point
+         */
         function plothover(event, pos, item) {
-            if(!that.stickyItem)
-            {
-                var $tip = that.getDomElement();
-                if (item) {
-                    var tipText;
+            if (item) {
+                const $tip = self.getDomElement();
+                const tooltipId = $tip.data('id');
+                const { tipPosition, isSticky } = self.tooltips[tooltipId];
 
+                if (!isSticky) {
                     // convert tooltip content template to real tipText
-                    tipText = that.stringFormat(that.tooltipOptions.content, item);
+                    const tipText = self.stringFormat(self.tooltipOptions.content, item);
 
                     $tip.html( tipText );
-                    that.updateTooltipPosition({ x: pos.pageX, y: pos.pageY });
+                    self.updateTooltipPosition({ x: pos.pageX, y: pos.pageY });
                     $tip.css({
-                        left: that.tipPosition.x + that.tooltipOptions.shifts.x,
-                        top: that.tipPosition.y + that.tooltipOptions.shifts.y
+                        left: tipPosition.x + self.tooltipOptions.shifts.x,
+                        top: tipPosition.y + self.tooltipOptions.shifts.y
                     }).show();
-
-                    // run callback
-                    if(typeof that.tooltipOptions.onHover === 'function') {
-                        that.tooltipOptions.onHover(item, $tip);
-                    }
                 }
-                else {
-                    hideTooltip();
+
+                if(typeof self.tooltipOptions.onHover === 'function') {
+                    self.tooltipOptions.onHover(item, $tip);
                 }
             }
+            else {
+                _handleHoverOnEmptyItem();
+            }
         }
 
         function plotclick(event, pos, item) {
-            var $tip = that.getDomElement();
-            if(that.tooltipOptions.stickyable) {
-                if(item && !that.stickyItem) {
+            var $tip = self.getDomElement();
```

Comment:
there is a bug here where i can have two sticky tooltips. Seems like we should unsticky the first tooltip when i try to make a second tooltip?

Reply:
I also got myself into a state where no tooltips were shown and i couldn't create any when  hovering or clicking on points.